src/Sylius/Bundle/OrderBundle/Form/Type/CartItemType.php

```diff
@@ -35,6 +44,27 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                 'attr' => ['min' => 1],
                 'label' => 'sylius.ui.quantity',
             ])
+            ->addEventListener(FormEvents::PRE_SUBMIT, function (FormEvent $event): void {
+                $newQuantity = $event->getData()['quantity'];
+                $orderItem = $event->getForm()->getData();
+                $oldQuantity = $orderItem->getQuantity();
+
+                if (!$oldQuantity) {
+                    return;
+                }
+
+                $variantId = $orderItem->getVariant()->getId();
+                $variant = $this->entityManager->getRepository(ProductVariant::class)->findOneBy(['id' => $variantId]);
```

Comment:
This variant is already there (there is even `$orderItem->getVariant()` above), why we should use a repository to get it again?